src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -877,12 +877,14 @@ export class GuidedTourService {
             return false;
         }
         const currentTourStep = this.currentTour.steps[this.currentTourStepIndex];
-        const selector = currentTourStep.highlightSelector;
-        if (selector) {
-            const selectedElement = document.querySelector(selector);
-            if (!selectedElement) {
-                return false;
-            }
+        if (!currentTourStep) {
+            this.resetTour();
+            return false;
+        }
+        const selector = currentTourStep.highlightSelector ? currentTourStep.highlightSelector : undefined;
+        const selectedElement = selector ? document.querySelector(selector) : undefined;
+        if (!selectedElement) {
+            return false;
         }
         return true;
```

Comment:
Is this actually the same as this:
```suggestion
        return currentTourStep.highlightSelector && document.querySelector(currentTourStep.highlightSelector);
```